geostructures/geohash.py

```diff
@@ -581,23 +588,35 @@ def _hash_polygon(
 
         return valid
 
-    def hash_collection(self, collection: ShapeCollection, **_):
+    def hash_collection(
+        self,
+        collection: ShapeCollection,
+        **kwargs
+    ) -> Dict[str, Any]:
         """
-        Hash a geostructures FeatureCollection. Returns a dictionary of hashes and
-        the corresponding number of shapes the hash has been observed in.
+        Hash a geostructures FeatureCollection. Returns a dictionary of hashes with
+        values equal to the output of an aggregation function over the shapes that
+        intersect each hash (by default, the number of shapes).
 
         Args:
             collection:
                 A geostructures FeatureCollection
 
+        Keyword Args:
+            agg_fn:
+                A function that accepts a list of geoshapes
+
         Returns:
-            dict
+            A dictionary of Niemeyer geohashes mapped to the result of the
+            aggregation function
         """
-        counter: Counter = Counter()
+        agg_fn = kwargs.get('agg_fn', len)
+        hash_dict: Dict[str, List[GeoShape]] = defaultdict(list)
         for shape in collection.geoshapes:
-            counter.update(self.hash_shape(shape))
-
-        return dict(counter)
+            shape_hashes = self.hash_shape(shape)
+            for hash in shape_hashes:

```

Comment:
Could you make this `for hash in self.hash_shape(shape)` just for cleanliness? Can remove the line above then